packages/legend-studio/src/components/editor/edit-panel/mapping-editor/PurePropertyMappingEditor.tsx

```diff
@@ -236,11 +238,20 @@ const ClassPropertyMappingEditor = observer(
     ) : (
       ''
     );
-    const expectedType =
+    let expectedType: Type | undefined;
+    if (
```

Comment:
nice! But like @MauricioUyaguari said, we could abtract this logic out to a function and use it for `relational` as well no?